extlinks/aggregates/management/commands/fill_link_aggregates.py

```diff
@@ -43,6 +44,11 @@ def handle(self, *args, **options):
             collections = Collection.objects.all().prefetch_related("url")
```

Comment:
Instead of checking for null while looping though objects, I suggest filtering those results out of the querysets. It should be more performant. Note that this is untested and just based on a read-through. I just picked this bit of code as an example, but it should apply elsewhere too.

```suggestion
            collections = Collection.objects.exclude(organisation__isnull=True) .prefetch_related("url")
```

Reply:
Yeah, that's better than the guard clause. I've added query arguments on all 6 places it was required.